camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+  private static final String MAPRED_COMPRESS_MAP_OUTPUT = "mapred.compress.map.output";
+  private static final boolean DEFAULT_MAPRED_COMPRESS_MAP_OUTPUT = Boolean.TRUE;
+  private static final String REDUCER_COUNT = "reducer.count";
+  private static final int DEFAULT_REDUCER_COUNT = 45;
+  private static final String MAPRED_MIN_SPLIT_SIZE = "mapred.min.split.size";
+  private static final String MAPRED_MAX_SPLIT_SIZE = "mapred.max.split.size";
+
+  static final String TOPIC_AND_HOUR = "topic.and.hour";
+  static final String INPUT_PATHS = "input.paths";
+  static final String DEST_PATH = "dest.path";
+
+  private final CamusSweeperMetrics metrics;
+
+  public CamusHourlySweeper() {
+    super();
+    metrics = new CamusSweeperMetrics();
+  }
+
+  public CamusHourlySweeper(Properties props) {
+    super(props);
+    metrics = new CamusSweeperMetrics();
+  }
+
+  @Override
+  public Map<FileStatus, String> findAllTopics(Path input, PathFilter filter, String topicSubdir, FileSystem fs)
+      throws IOException {
+    Map<FileStatus, String> topics = new HashMap<FileStatus, String>();
+    for (FileStatus f : fs.listStatus(input)) {
+      if (f.isDir() && filter.accept(f.getPath())) {
+        LOG.info("found topic: " + f.getPath().getName());
+        topics.put(fs.getFileStatus(f.getPath()), f.getPath().getName());
+      }
+    }
+    return topics;
+  }
+
+  @Override
+  public void run() throws Exception {
+    metrics.timeStart = System.currentTimeMillis();
+    super.run();
+    reportMetrics();
+    addOutliers();
+  }
+
+  private void addOutliers() throws IOException {
+    Configuration conf = new Configuration();
+    for (Entry<Object, Object> pair : props.entrySet()) {
+      String key = (String) pair.getKey();
+      conf.set(key, (String) pair.getValue());
+    }
+    for (Properties jobProps : planner.getOutlierProperties()) {
+      this.executorService.submit(new OutlierCollectorRunner(jobProps, FileSystem.get(conf)));
+    }
+  }
+
+  private void reportMetrics() throws IllegalArgumentException, IOException {
+    LOG.info("reporting metrics");
+    metrics.reportTotalRunningTime();
+    metrics.reportTotalDataSize();
+    metrics.reportDataSizeByTopic();
+    metrics.reportDurationFromStartToRunnerStart();
+    metrics.reportDurationFromRunnerStartToMRSubmitted();
+    metrics.reportDurationFromMRSubmittedToMRStarted();
+    metrics.reportDurationFromMRStartedToMRFinished();
+    LOG.info("finished reporting metrics");
+  }
+
+  public static void main(String args[]) throws Exception {
+    CamusSweeper job = new CamusHourlySweeper();
+    ToolRunner.run(job, args);
+  }
+
+  @Override
+  protected Future<?> runCollector(Properties props, String topic) {
+    String jobName = topic + "-" + UUID.randomUUID().toString();
+    props.put("tmp.path", props.getProperty(CAMUS_SWEEPER_TMP_DIR) + "/" + jobName + "_" + System.currentTimeMillis());
```

Comment:
It's fragile to construct path by concatenating path components, due to absence/presence of the trailing slash. Better to do `new Path(props.getProperty(CAMUS_SWEEPER_TMP_DIR), jobName + "_" + System.currentTimeMillis()).toUri().getPath()`.